### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by cs-sujeet-ahirrao**
- Removed `TAXII Outgoing Feed` support 

- Added new Threat Feed Rules
    - `Unstructured Threat Feeds Support`
    - `Block High Confidence Threat Feeds`

- Updated `Threat Feed To Indicator Linking` threat feed rule
    - `Feed Confidence Threshold` is also updated in Key Store Record and global variable

- Unit Test Cases 
    - At load of widget check `Email Ingestion Schedule` and `Block Threat Feeds Automatically` schedules are exist or not. If not then create the schedule.
    - If `Feed Confidence Threshold` is updated in the `Threat Feed To Indicator Linking` threat feed rule then the value is updated in key store record and global variable
    - Checked all the validation for all the fields are working as expected
    - Default/Updated values are pre populated from the key store record in specified fields while widget loading
    - Checked if any field value is updated then that value also updated in key store Record

### CODE REVIEW

**Review by cs-pranali-chavan — APPROVED** [review]
Looks good to me

File: widget/widgetAssets/html/feedConfigSchedule.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
+<!-- Copyright start
+  MIT License
+  Copyright (c) 2024 Fortinet Inc
+  Copyright end -->
+<form data-ng-submit="save()" class="noMargin" name="scheduleForm" data-ng-class="{'state-wait': processing }"
+  novalidate>
+  <fieldset data-ng-disabled="!allowEdit">
+    <div class="row">
+      <div data-ng-if="!params.form.schedule.kwargs.exit_if_running"
+        class="col-md-12 font-size-14 control-label margin-bottom-md padding-0">Do you want to schedule the ingestion?<span
```
Comment (cs-suyog-jinturkar) [inline widget/widgetAssets/html/feedConfigSchedule.html:10]: i18n changes?

File: widget/widgetAssets/js/feedConfigSchedule.directive.js [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+      });
+
+      function updateCron(updatedCronSection) {
+        var cronstrue = $window.cronstrue;
+        scope.cronDescriber = '';
+        if (scope.scheduleConfig.crontab.minute !== '' && scope.scheduleConfig.crontab.hour !== '' && scope.scheduleConfig.crontab.day_of_month !== '' && scope.scheduleConfig.crontab.month_of_year !== '' && scope.scheduleConfig.crontab.day_of_week !== '') {
+          scope.cronDescriber = cronstrue.toString(scope.scheduleConfig.crontab.minute + ' ' + scope.scheduleConfig.crontab.hour + ' ' + scope.scheduleConfig.crontab.day_of_month + ' ' + scope.scheduleConfig.crontab.month_of_year + ' ' + scope.scheduleConfig.crontab.day_of_week);
+        }
+        scope.params.config.cronName = angular.copy(scope.cronDescriber);
+        //scope.$emit('feedScheduleDetails', { 'status': true, 'scheduleDetails': scope.scheduleConfig, 'scheduleFrequency': scope.cronDescriber });
```
Comment (cs-suyog-jinturkar) [inline widget/widgetAssets/js/feedConfigSchedule.directive.js:121]: commented code, do we need it?

File: widget/view.html [hunk 3]
```diff
@@ -3,229 +3,489 @@
...
-                                <tr class="body" data-ng-if="!datasetProcessing && datasets.length === 0">
-                                  <td colspan="2" class="muted font-size-12 text-center">No datasets found.</td>
-                                </tr>
-                              </table>
-                            </div>
-                          </div>
+  <div data-ng-show="getKeyStoreRecord" class="row position-relative full-height">
+    <form name="feedConfigurationSettingsForm" class="feed-configuration-container">
+      <div class="col-md-7 threat-feed-config-uib-accordion padding-top-30">
+        <div class="font-size-25 muted-80 line-height-1 padding-bottom-md">Threat Feed Rules</div>
```
Comment (cs-suyog-jinturkar) [inline widget/view.html:16]: i18n compliant?

File: widget/view.controller.js [hunk 4]
```diff
@@ -1,233 +1,335 @@
...
+
+    function updatesendNotification() {
+      if (!$scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromFile.enabled && !$scope.feedRules.unstructuredFeedsSupport.ingestFeedsFromEmail.enabled) {
+        $scope.feedRules.unstructuredFeedsSupport.sendEmailNotifications.enabled = false;
+      }
+    }
+
+    function getEmailClientDetails(name) {
+      var queryBody = {
+        "sort": [
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:127]: please move this to the service code

File: widget/view.controller.js [hunk 5]
```diff
@@ -1,233 +1,335 @@
...
+
+    function _createSchedule(scheduleName, scheduleDetails, crontab) {
+      var queryBody = {
+        name: scheduleDetails.name,
+        "crontab": crontab,
+        "kwargs": {
+          "exit_if_running": true,
+          "wf_iri": API.API_3_BASE + API.WORKFLOWS + scheduleDetails.playbook_uuid,
+          "timezone": "UTC",
+          "utcOffset": "UTC",
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:157]: this is a reusable code, this can be moved to service code.

File: widget/view.controller.js [hunk 6]
```diff
@@ -1,233 +1,335 @@
...
+            // If the schedule is not enabled, activate it
+            if (!$scope.saveSchedules[scheduleName].enabled) {
+              $scope.saveSchedules[scheduleName].enabled = true;
+              _updateSchedule($scope.saveSchedules[scheduleName]);
             }
+          });
+        } else {
+          // If schedule exists, use the existing one
+          $scope.saveSchedules[scheduleName] = {
+            id: response['hydra:member'][0].id,
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:188]: repeated code, can be optimised and moved into one function.

File: widget/view.controller.js [hunk 7]
```diff
@@ -1,233 +1,335 @@
...
-                  $scope.oldTaxiiServerStatusObject.publicValues.status.enabled;
-                $scope.params.feedOutgoingStatusForm.$setPristine();
-                toaster.success({
-                  body: "Changes reverted successfully.",
-                });
-              }
-            });
-          }
+    function _getFeedRulesSettings() {
+      return $resource(API.WORKFLOW + "api/dynamic-variable/?name=Indicator_Feed_Reputation_Preference&format=json").get({}).$promise.then(function (response) {
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:220]: can be moved to service code

File: widget/view.controller.js [hunk 8]
```diff
@@ -1,233 +1,335 @@
...
+        $scope.feedConfidenceThresholdVariable = response["hydra:member"][0];
+        var parsedFeedConfigValues = JSON.parse($scope.feedConfidenceThresholdVariable.value);
+        $scope.feedRules.feedToIndicatorLinking.feedConfidenceThreshold = parsedFeedConfigValues.confidenceThreshold;
+        $scope.feedRules.feedToIndicatorLinking.enabled =
+          parsedFeedConfigValues.setReputation;
+      }, statusCodeService);
+    }
+
+    function _updateSchedule(scheduleDetails) {
+      var url = API.WORKFLOW + 'api/scheduled/' + scheduleDetails.id + '/?format=json';
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:230]: same as above

File: widget/view.controller.js [hunk 9]
```diff
@@ -1,233 +1,335 @@
...
+          console.log("Schedule updated successfully.");
+        }, function () {
+          toaster.error({
+            body: "Unable to update schedule.",
+          });
+        });
+    }
+
+    function _saveFeedToIndicatorLinking(feedToIndicatorLinkingForm) {
+      $scope.feedConfidenceThresholdVariable.value = JSON.stringify({
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:244]: same as above

File: widget/view.controller.js [hunk 10]
```diff
@@ -1,233 +1,335 @@
...
-                  body: "Changes reverted successfully.",
-                });
-              }
-            });
+      if (!_.isEqual($rootScope.oldBlockHighConfidenceThreatFeeds, $scope.feedRules.blockHighConfidenceThreatFeeds)) {
+        feedConfigurationSettingsForm.blockHighConfidenceThreatFeedsForm.$setPristine();
+        feedConfigurationSettingsForm.blockHighConfidenceThreatFeedsForm.$dirty = false;
+
+        if (!$scope.feedRules.blockHighConfidenceThreatFeeds.enabled || !$scope.feedRules.blockHighConfidenceThreatFeeds.automaticBlockIOCEnabled.enabled) {
+          if ($scope.feedRules.blockHighConfidenceThreatFeeds.automaticBlockIOCEnabled.threatFeedBlockSchedule.avtivateScheduleQueryParams.enabled !== false) {
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:284]: seems to be repeated code. Can be optimised futher

File: widget/widgetAssets/html/feedConfigSchedule.html (hunk 1 above)
Comment (cs-sujeet-ahirrao) [reply widget/widgetAssets/html/feedConfigSchedule.html:10]: will add in final checkin

File: widget/view.html (hunk 3 above)
Comment (cs-sujeet-ahirrao) [reply widget/view.html:16]: same as above

File: widget/widgetAssets/js/feedConfigSchedule.directive.js (hunk 2 above)
Comment (cs-sujeet-ahirrao) [reply widget/widgetAssets/js/feedConfigSchedule.directive.js:121]: Will remove in Final build

File: widget/view.controller.js (hunk 5 above)
Comment (cs-sujeet-ahirrao) [reply widget/view.controller.js:157]: USed in chain promise so can't move to service code

**Pull request merged**